tests/test_schemes.py

```diff
@@ -103,3 +104,43 @@ def test_concatenated_scheme():
                                       ConstantScheme(batch_size=30, times=1)])
     assert (list(sch.get_request_iterator()) ==
             ([10] * 5) + ([20] * 3) + [30])
+
+
+def test_k_fold_cross_validation():
+    # test raise when strict=True
+    cross = k_fold_cross_validation(SequentialExampleScheme, 10, 3)
+    assert_raises(ValueError, next, cross)
+
+    # test BatchScheme when strict=False
```

Comment:
I think there's a typo: you're testing an `IndexScheme` here and a `BatchScheme` further down.